Timestamp: 12/23/07 18:42:57
Author: courmisch
Fix format string injection in the Connection field for HTTP-served files
(affects HTTP interface and LUA plus HTTP export for DVB and RTP) 

Index: vlc-0.8.6d/src/network/httpd.c
===================================================================
--- vlc-0.8.6d.orig/src/network/httpd.c
+++ vlc-0.8.6d/src/network/httpd.c
@@ -414,7 +414,7 @@ static int httpd_FileCallBack( httpd_cal
     psz_connection = httpd_MsgGet( &cl->query, "Connection" );
     if( psz_connection != NULL )
     {
-        httpd_MsgAdd( answer, "Connection", psz_connection );
+        httpd_MsgAdd( answer, "Connection", "%s", psz_connection );
     }
 
     httpd_MsgAdd( answer, "Content-Length", "%d", answer->i_body );
